include/sparrow/fixed_size_layout.hpp

```diff
@@ -0,0 +1,369 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <algorithm>
+#include <concepts>
+#include <cstdint>
+#include <cassert>
+
+#include "sparrow/array_data.hpp"
+#include "sparrow/iterator.hpp"
+#include "sparrow/buffer.hpp"
+#include "sparrow/dynamic_bitset.hpp"
+
+namespace sparrow
+{
+    /**
+     * An iterator for `fixed_size_layout` operating on contiguous data only.
+     *
+     * @tparam T The type of the elements in the layout's data buffer.
+     * @tparam is_const A boolean indicating whether the iterator is const.
+     *
+     * @note This class is copyable and movable.
```

Comment:
Note sure why you specified that? It's the default/usual/regular/normal/expected in C++.

Reply:
Let me remove such notes.